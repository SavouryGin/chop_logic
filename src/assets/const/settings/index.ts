import { SelectEntity } from 'types';

export const languageOptions: SelectEntity[] = [
  { option: { en: 'English', ru: 'Английский' }, value: 'en' },
  { option: { en: 'Russian', ru: 'Русский' }, value: 'ru' },
];

export const externalLinks: { [key in string]: string } = {
  telegram: 'https://telegram.me/savoury_gin',
  gitHub: 'https://github.com/SavouryGin',
  mail: 'mailto:Dmitrii.Suroviagin@gmail.com',
};
