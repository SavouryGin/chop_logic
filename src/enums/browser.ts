export enum Browser {
  Unknown,
  Opera,
  Chrome,
  Safari,
  Firefox,
}
