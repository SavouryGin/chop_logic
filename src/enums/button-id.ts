export enum ButtonID {
  Cancel,
  Navigation,
  ColorTheme,
  Sounds,
  Settings,
  FullScreen,
  Tools,
  Apply,
  Premise,
  Reiteration,
  Replace,
  Delete,
  ImplicationCreation,
  ImplicationDistribution,
  ContradictionRealization,
  ImplicationElimination,
  Assumption,
  NotIntroduction,
  NotElimination,
  AndIntroduction,
  AndElimination,
  OrIntroduction,
  OrElimination,
  ImpliesIntroduction,
  ImpliesElimination,
  EquivIntroduction,
  EquivElimination,
  Shortcut,
  SavePDF,
  ExportXML,
  ImportXML,
  CopyProof,
  PasteProof,
  CutProof,
  Reset,
}
