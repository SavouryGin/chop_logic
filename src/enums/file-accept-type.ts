export enum FileAcceptType {
  Image = 'image/*',
  Video = 'video/*',
  Audio = 'audio/*',
  XML = 'text/xml',
  PDF = 'application/pdf',
  CSV = '.csv',
  Text = 'text/plain',
  Excel = 'application/vnd.ms-excel, application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
}
