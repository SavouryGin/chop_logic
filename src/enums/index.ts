export enum Icon {
  Default = 'chop-icon__check',
  Home = 'chop-icon__home',
  Menu = 'chop-icon__menu',
  Enlarge = 'chop-icon__enlarge2',
  Shrink = 'chop-icon__shrink2',
  Settings = 'chop-icon__cog',
  Delete = 'chop-icon__bin2',
  Up = 'chop-icon__circle-up',
  Down = 'chop-icon__circle-down',
  Right = 'chop-icon__circle-right',
  Left = 'chop-icon__circle-left',
  Sound = 'chop-icon__volume-medium',
  NoSound = 'chop-icon__volume-mute2',
  LightMode = 'chop-icon__sun',
  DarkMode = 'chop-icon__contrast',
}

export enum Pages {
  Home,
  NotFound,
  Propositions,
  Predicates,
  TruthTables,
  Syllogisms,
}
