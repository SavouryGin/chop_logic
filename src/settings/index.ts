import { SelectEntity } from 'types';

export const languageOptions: SelectEntity[] = [
  { option: { en: 'English', ru: 'Английский' }, value: 'en' },
  { option: { en: 'Russian', ru: 'Русский' }, value: 'ru' },
];

export const languageStringOptions = ['en', 'ru'];

export const externalLinks: { [key in string]: string } = {
  telegram: 'https://telegram.me/savoury_gin',
  gitHub: 'https://github.com/SavouryGin',
  linkedIn: 'https://www.linkedin.com/in/dmitrii-suroviagin/',
  mail: 'mailto:Dmitrii.Suroviagin@gmail.com',
  facebook: 'https://www.facebook.com/dmitrii.suroviagin',
};
