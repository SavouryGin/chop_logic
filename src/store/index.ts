import createSagaMiddleware from 'redux-saga';
import rootReducer from './root-reducer';
import rootSaga from './root-saga';
import { configureStore } from '@reduxjs/toolkit';

const sagaMiddleware = createSagaMiddleware();

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat([sagaMiddleware]),
});

sagaMiddleware.run(rootSaga);
