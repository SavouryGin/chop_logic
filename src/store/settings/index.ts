import { Language } from 'types';
import { PayloadAction, createSlice } from '@reduxjs/toolkit';
import { SettingsFlag, SettingsInitialState } from './interfaces';

export const settingsInitialState: SettingsInitialState = {
  language: 'en',
  flags: {
    isNavigationOpen: false,
    isSidebarOpen: false,
    isDarkMode: false,
    isFullScreen: false,
    isSettingOpen: false,
    isSoundsEnabled: false,
  },
};

export const settingsSlice = createSlice({
  name: 'settings',
  initialState: settingsInitialState,
  reducers: {
    resetSettings: () => {
      return settingsInitialState;
    },

    toggleFlag: (state, action: PayloadAction<SettingsFlag>) => {
      const currentFlagValue = state.flags[action.payload];
      state.flags[action.payload] = !currentFlagValue;
    },

    setLanguage: (state, action: PayloadAction<Language>) => {
      state.language = action.payload;
    },

    setFullScreenFlag: (state, action: PayloadAction<boolean>) => {
      state.flags.isFullScreen = action.payload;
    },

    setUpFlag: (state, action: PayloadAction<{ flag: SettingsFlag; value: boolean }>) => {
      const { flag, value } = action.payload;
      state.flags[flag] = value;
    },
  },
});

export const settingsActions = { ...settingsSlice.actions };
