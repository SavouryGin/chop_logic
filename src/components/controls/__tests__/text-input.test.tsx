import React from 'react';
import TextInput from 'components/controls/text-input';
import renderWithRedux from 'utils/testing/render-with-redux';
import userEvent from '@testing-library/user-event';
import { InputID } from 'enums';
import { combineReducers } from '@reduxjs/toolkit';
import { screen } from '@testing-library/react';
import { settingsInitialState, settingsSlice } from 'store/settings';

describe('Text input component:', () => {
  const testProps = {
    name: 'test input',
    label: 'test label',
    inputId: InputID.DefaultInput,
  };

  const mockedReducer = combineReducers({ settings: settingsSlice.reducer });
  const mockedState = { settings: settingsInitialState };

  it('renders the textbox element with default props', () => {
    renderWithRedux(<TextInput {...testProps} />, mockedReducer, mockedState);
    const input = screen.getByRole('textbox');
    expect(input).toBeInTheDocument();
    expect(input).toHaveClass('text-input__field');
    expect(input).toHaveValue('');
    expect(input).toBeEnabled();
  });

  it('the input has a label', () => {
    renderWithRedux(<TextInput {...testProps} />, mockedReducer, mockedState);
    expect(screen.getByLabelText(testProps.label)).toBeInTheDocument();
  });

  it('receives the passed default value', () => {
    renderWithRedux(<TextInput {...testProps} defaultValue='test-default' />, mockedReducer, mockedState);
    expect(screen.getByRole('textbox')).toHaveValue('test-default');
  });

  it('can be disabled', () => {
    renderWithRedux(<TextInput {...testProps} isDisabled />, mockedReducer, mockedState);
    expect(screen.getByRole('textbox')).toBeDisabled();
  });

  it('receives the length properties', () => {
    renderWithRedux(<TextInput {...testProps} maxLength={100} minLength={10} />, mockedReducer, mockedState);
    const input = screen.getByRole('textbox');
    expect(input).toHaveProperty('maxLength', 100);
    expect(input).toHaveProperty('minLength', 10);
  });

  it('displays the passed placeholder for the input', () => {
    renderWithRedux(<TextInput {...testProps} placeholder='test placeholder' />, mockedReducer, mockedState);
    expect(screen.getByPlaceholderText('test placeholder')).toBeInTheDocument();
  });

  it('allows the user to type some text', async () => {
    renderWithRedux(<TextInput {...testProps} />, mockedReducer, mockedState);
    const input = screen.getByRole('textbox');
    await userEvent.type(input, 'test text');
    expect(input).toHaveValue('test text');
  });

  it('fires the passed onChange handler', async () => {
    const mockChange = jest.fn();
    renderWithRedux(<TextInput {...testProps} onChange={mockChange} />, mockedReducer, mockedState);
    const input = screen.getByRole('textbox');
    await userEvent.type(input, '123');
    expect(mockChange).toHaveBeenCalledTimes(3);
  });

  it('gets focus on tab press', async () => {
    renderWithRedux(<TextInput {...testProps} />, mockedReducer, mockedState);
    const input = screen.getByRole('textbox');
    expect(input).not.toHaveFocus();
    await userEvent.tab();
    expect(input).toHaveFocus();
  });

  it('fires the onBlur handler', async () => {
    const mockBlur = jest.fn();
    renderWithRedux(<TextInput {...testProps} onBlur={mockBlur} />, mockedReducer, mockedState);
    await userEvent.tab();
    await userEvent.tab();
    expect(mockBlur).toHaveBeenCalledTimes(1);
  });

  it('fires the onFocus handler', async () => {
    const mockFocus = jest.fn();
    renderWithRedux(<TextInput {...testProps} onFocus={mockFocus} />, mockedReducer, mockedState);
    await userEvent.tab();
    expect(mockFocus).toHaveBeenCalledTimes(1);
  });

  it('should match the snapshot', () => {
    const { asFragment } = renderWithRedux(<TextInput {...testProps} />, mockedReducer, mockedState);
    expect(asFragment()).toMatchSnapshot();
  });
});
