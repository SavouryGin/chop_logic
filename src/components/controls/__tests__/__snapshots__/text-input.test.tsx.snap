// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text input component: should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-input"
  >
    <label
      class="label"
      data-testid="label-test-id"
      for="text_input_id_0"
    >
      <span
        class="label__text"
      >
        test label
      </span>
    </label>
    <input
      autocomplete="off"
      class="text-input__field"
      id="text_input_id_0"
      name="test input"
      type="text"
      value=""
    />
    <button
      class="text-input__clear chop-icon__cross text-input__clear_hidden"
      type="reset"
    />
  </div>
</DocumentFragment>
`;
